test/testkit/matchers/trace/ptrace_utils.go

```diff
@@ -1,30 +1,61 @@
 package trace
 
 import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 
 	"github.com/kyma-project/telemetry-manager/test/testkit/matchers"
 )
 
+type FlatTrace struct {
+	Name                               string
+	ResourceAttributes, SpanAttributes map[string]string
+}
+
 func unmarshalTraces(jsonlMetrics []byte) ([]ptrace.Traces, error) {
 	return matchers.UnmarshalSignals[ptrace.Traces](jsonlMetrics, func(buf []byte) (ptrace.Traces, error) {
 		var unmarshaler ptrace.JSONUnmarshaler
 		return unmarshaler.UnmarshalTraces(buf)
 	})
 }
 
-func getSpans(td ptrace.Traces) []ptrace.Span {
-	var spans []ptrace.Span
+func flattenAllTraces(tds []ptrace.Traces) []FlatTrace {
+	var flatTraces []FlatTrace
+
+	for _, td := range tds {
+		flatTraces = append(flatTraces, flattenTraces(td)...)
+	}
+	return flatTraces
+}
+
+func flattenTraces(td ptrace.Traces) []FlatTrace {
+	var flatTraces []FlatTrace
 
 	for i := 0; i < td.ResourceSpans().Len(); i++ {
 		resourceSpans := td.ResourceSpans().At(i)
 		for j := 0; j < resourceSpans.ScopeSpans().Len(); j++ {
 			scopeSpans := resourceSpans.ScopeSpans().At(j)
 			for k := 0; k < scopeSpans.Spans().Len(); k++ {
-				spans = append(spans, scopeSpans.Spans().At(k))
+				spans := scopeSpans.Spans().At(k)
+				for l := 0; l < spans.Attributes().Len(); l++ {
```

Comment:
why are you looping here over the span attributes?
check the [contexts](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/transformprocessor/README.md#contexts) here.

-  A Metric contains Datapoints
- A Span contains SpanEvents

If you want to have something similar to what we have in the FlatMetrics, then you should loop over the SpanEvents in the Span and not its attributes. However, do we need any fields from the SpanEvents in our e2e tests so far? If not then, you don't need to loop over them.